lib/services/user_service.dart

```diff
@@ -90,6 +90,31 @@ class UserService {
     }
   }
 
+  Future<void> logout(BuildContext context) async {
+    final dialog = createProgressDialog(context, "logging out...");
+    await dialog.show();
+    try {
+      final response =
+          await _dio.post(_config.getHttpEndpoint() + "/users/logout",
+              options: Options(
+                headers: {
+                  "X-Auth-Token": _config.getToken(),
+                },
+              ));
+      if (response != null && response.statusCode == 200) {
+        await Configuration.instance.logout();
+        await dialog.hide();
+        Navigator.of(context).popUntil((route) => route.isFirst);
+      } else {
+        throw new Exception("log out action failed");
```

Comment:
Would need to hide `dialog` here too? Also, could just log the exception here and show the generic error dialog.

Reply:
> Would need to hide `dialog` here too? Also, could just log the exception here and show the generic error dialog.

As discussed offline, it's handled in the catch block. Ideally, we should never reach here.